File: evidence/generated-systems/golang/advertising-on-golang/advertising-on-golang/internal/model/Publisher.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Publisher Declaration
//==============================================================
type Publisher struct {
    gorm.Model
     Name                                    string
    Website                                    string
     InventorySources           []InventorySource `gorm:"foreignKey:InventorySourcesFromPublisherId"`
     Deals           []Deal `gorm:"foreignKey:DealsFromPublisherId"`
     CreativeApprovals           []CreativeApproval `gorm:"foreignKey:CreativeApprovalsFromPublisherId"`
     InsertionOrders           []InsertionOrder `gorm:"foreignKey:InsertionOrdersFromPublisherId"`
     RateCards           []RateCard `gorm:"foreignKey:RateCardsFromPublisherId"`
    PublisherType                      PublisherType

// parent associations as their child

}

File: evidence/generated-systems/angular/advertising-on-angular/src/app/models/Publisher.js

// Define collection and schema for Publisher
export interface Publisher {
    name:
	type : string
    website:
	type : string
    InventorySources:
 	type : [{ type: Schema.Types.ObjectId, ref: 'InventorySource' }]
    Deals:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Deal' }]
    CreativeApprovals:
 	type : [{ type: Schema.Types.ObjectId, ref: 'CreativeApproval' }]
    InsertionOrders:
 	type : [{ type: Schema.Types.ObjectId, ref: 'InsertionOrder' }]
    RateCards:
 	type : [{ type: Schema.Types.ObjectId, ref: 'RateCard' }]
    PublisherType:
 	type : String
#
    collection: 'publishers'
}
